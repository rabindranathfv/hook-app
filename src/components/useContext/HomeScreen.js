import React, { useContext } from 'react'
import { UserContext } from './UserContest'

export const HomeScreen = () => {

    const { user, setUser } = useContext(UserContext)

    return (
        <div>
           <h1> Home Screen </h1>
            <hr /> 

            <pre> 
                { JSON.stringify(user, null, 3) }
            </pre>
        </div>
    )
}
