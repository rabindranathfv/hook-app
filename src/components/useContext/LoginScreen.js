import React, { useContext, useEffect } from 'react';
import { UserContext } from './UserContest';

export const LoginScreen = () => {
    
    const { user, setUser } = useContext(UserContext);

    useEffect(() => {
        console.log('Listem all user changes')
    }, [user])
    

    const handleInputChange = (e) => {
        console.log(user);
        setUser({
            ...user,
            [ e.target.name ] : e.target.value
    
        })
    };

    return (
        <div>
            <h1> Login Screen </h1>
            <hr />

            <div className="form-group">
                <label htmlFor=""></label>
                <input type="text"
                className="form-control" name="name" id="" aria-describedby="name-form" placeholder="enter your name" value={user?.name} onChange={handleInputChange}/>
                <small className="username info">Name</small>

                <label htmlFor=""></label>
                <input type="text"
                className="form-control" name="email" id="" aria-describedby="email-form" placeholder="enter your email" value={user?.email} onChange={handleInputChange}/>
                <small className="email info">Email</small>
            </div>
        </div>
    )
}
