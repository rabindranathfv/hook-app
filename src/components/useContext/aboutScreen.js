import React, { useContext } from 'react'
import { UserContext } from './UserContest';

export const AboutScreen = () => {

    const { user, setUser } = useContext(UserContext);

    return (
        <div>
            <h1> About Screen </h1>
            <hr />

            <h3>
                welcome { user?.name }!
            </h3>
        </div>
    )
}
