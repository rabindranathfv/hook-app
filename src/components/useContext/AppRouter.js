import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Routes,
    useRouteMatch,
    useParams
} from 'react-router-dom';

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <NavBar />

                <Routes>
                    <Route path="/about" element={<AboutScreen />} />
                    <Route path="/login" element={<LoginScreen />} />
                    <Route path="/" element={<HomeScreen />} />
                    <Route path="*" element={<HomeScreen />} />
                </Routes>
            </div>
        </Router>
    )
}
