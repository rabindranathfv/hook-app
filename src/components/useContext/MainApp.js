import React, { useState } from 'react'
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContest';


export const MainApp = () => {
    
    const [user, setUser] = useState({ name: '', email: ''})
    
    return (
        <UserContext.Provider value={ { user, setUser } }>
            <AppRouter />
        </UserContext.Provider>
    )
}
