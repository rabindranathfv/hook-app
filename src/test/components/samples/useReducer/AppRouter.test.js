import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from './../../../../components/useContext/AppRouter';
import { UserContext } from '../../../../components/useContext/UserContest';

const user = {
    name: 'rabindranath',
    email: 'test@gmail.com'
}
describe('AppRouter', () => {

    let wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <AppRouter />
        </UserContext.Provider>
    );

    beforeEach( () => {
        wrapper= mount(
            <UserContext.Provider value={{ user }}>
                <AppRouter />
            </UserContext.Provider>
        );
    })
  it('Should render AppRouter component', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it('Should render navBar component', () => {
      const navBarComponent = wrapper.find('Navbar');

      expect(navBarComponent).toBeDefined();
  });


});
