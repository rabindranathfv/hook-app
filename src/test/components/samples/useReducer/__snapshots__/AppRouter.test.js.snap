// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter Should render AppRouter component 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
         Home Screen 
      </h1>
      <hr />
      <pre>
        {
   "name": "rabindranath",
   "email": "test@gmail.com"
}
      </pre>
    </div>
  </div>
</div>
`;
